app/src/main/java/com/mattgmg/miracastwidget/MainActivity.java

```diff
@@ -53,16 +37,11 @@ protected void onCreate(Bundle savedInstanceState) {
                 systemWifiIntent.setClassName(systemResolveInfo.activityInfo.applicationInfo.packageName,
                                     systemResolveInfo.activityInfo.name);
                 startSettingsActivity(systemWifiIntent);
-                sendEvent(CATEGORY_CAST_LAUNCH, ACTION_SUCCESS, "wifi_action");
                 finish();
                 return;
             } catch (ActivityNotFoundException exception) {
                 // We'll show an error below if the next Intent can't be launched
-                mTracker.send(new HitBuilders.ExceptionBuilder()
-                                .setDescription("Launching systemResolveInfo for wifi action")
-                                .setFatal(false)
-                                .build());
-                sendEvent(CATEGORY_CAST_LAUNCH, ACTION_ERROR, "wifi_action_ANF_exception");
+                exception.printStackTrace();
```

Comment:
We should either log to logcat with Log.e or just remove this. See https://stackoverflow.com/questions/3855187/is-it-a-bad-idea-to-use-printstacktrace-for-caugt-exceptions

Can you just remove the exception printing everywhere please? If we're catching and we don't need to catch for any reason other than to log, we can just remove the try/catch.

Reply:
I'm sorry, I guess I just didn't notice using ``printStackTrace()`` at that moment.

> If we're catching and we don't need to catch for any reason other than to log, we can just remove the try/catch.

Maybe I've misunderstood you, but why would you want to remove ``try/catch`` when ``catch`` does nothing? Isn't that better than crashing the whole app, especially when some exceptions are device-dependent/only thrown on devices that don't support Miracast? Also, the method signature of ``onCreate()`` doesnt allow for throwing any non-runtime exceptions like ``PackageManager.NameNotFoundException``.

I've just removed exception logging for now. ``try/catch`` seems to serve another purpose besides preventing the app from crashing: When Miracast is not supported, a 🍞 is shown. 🙂

Reply:
Oh yes, of course. Sorry, I wasn't really thinking when I wrote that comment. You're right, we don't want to crash. Sure, just leaving it sounds good.